src/lib/__tests__/insights-client-test.ts

```diff
@@ -0,0 +1,261 @@
+import { withInsights, inferInsightsPayload } from '../insights';
+
+const connectHits = (renderFn: any, unmountFn: any) => (widgetParams = {}) => ({
+  init() {},
+  render({ results, instantSearchInstance }) {
+    const hits = results.hits;
+    renderFn({ hits, results, instantSearchInstance, widgetParams }, false);
+  },
+  dispose() {
+    unmountFn();
+  },
+});
+
+const setup = ({ hits: providedHits, insightsClient }: any = {}) => {
+  const instantSearchInstance = {
+    insightsClient: insightsClient || jest.fn(),
+  };
+  const hits = providedHits || [
+    { objectID: '1', __queryID: 'theQueryID', __position: 9 },
+    { objectID: '2', __queryID: 'theQueryID', __position: 10 },
+    { objectID: '3', __queryID: 'theQueryID', __position: 11 },
+    { objectID: '4', __queryID: 'theQueryID', __position: 12 },
+  ];
+  const results = {
+    index: 'theIndex',
+    hits,
+  };
+  const widgetParams = { transformItems: jest.fn() };
+
+  const renderFn = jest.fn();
+  const unmountFn = jest.fn();
+
+  const connectHitsWithInsights = withInsights(connectHits as any);
+  const widget = connectHitsWithInsights(renderFn, unmountFn)(widgetParams);
+  widget.render({ results, instantSearchInstance } as any);
+  return { widget, renderFn, instantSearchInstance, results, widgetParams };
+};
+
+describe('withInsights', () => {
+  describe('when applied on connectHits', () => {
+    it('should call the passed renderFn', () => {
+      const { renderFn } = setup();
+      expect(renderFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not remove any renderProps passed by connectHits', () => {
+      const {
+        renderFn,
+        instantSearchInstance,
+        results,
+        widgetParams,
+      } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toEqual(
+        expect.objectContaining({
+          instantSearchInstance,
+          results,
+          widgetParams,
+        })
+      );
+    });
+
+    it('should expose the insights client wrapper to renderOptions if passed to instantSearchInstance', () => {
+      const { renderFn } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+
+    it('should not expose the insights client wrapper to renderOptions if not passed to instantSearchInstance', () => {
+      const { renderFn } = setup({ insightsClient: undefined });
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+  });
+
+  describe('exposed insights client wrapper', () => {
+    beforeEach(() => {});
```

Comment:
We can drop it since it's useless.

Reply:
fixed in ce61df08bf0278a0c3cbca46c9ad3e17cb490c87